drivers/digital-io/max22200/iio_max22200.c

```diff
@@ -0,0 +1,1091 @@
+};
+
+static struct iio_attribute max22200_global_attrs[] = {
+	{
+		.name = "chopping_frequency",
+		.show = max22200_iio_read_chopping_freq,
+		.store = max22200_iio_write_chopping_freq,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "chopping_frequency_available",
+		.show = max22200_iio_read_available,
+		.priv = MAX22200_IIO_CHOP_FREQ_AVAILABLE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "external_trig_state",
+		.store = max22200_iio_write_trig_state,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_device max22200_iio_dev = {
+	.debug_reg_read = (int32_t (*)())max22200_iio_reg_read,
+	.debug_reg_write = (int32_t (*)())max22200_iio_reg_write,
+	.debug_attributes = max22200_debug_attrs,
+	.attributes = max22200_global_attrs
+};
+
+/**
+ * @brief Read the raw attribute for a specific channel (state of the channel)
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	bool ch_state;
+	int32_t val;
+	int ret;
+
+	ret = max22200_get_ch_state(desc->max22200_desc, channel->address,
+				    &ch_state);
+	if (ret)
+		return ret;
+
+	val = ch_state ? 1 : 0;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+/**
+ * @brief Read the scale attribute for a specific channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_read_scale(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_scale ch_scale;
+	int ret;
+
+	ret = max22200_get_ch_scale(desc->max22200_desc, channel->address,
+				    &ch_scale);
+	if (ret)
+		return ret;
+
+	strcpy(buf, max22200_scale_avail[ch_scale]);
+
+	return strlen(buf);
+}
+
+/**
+ * @brief Read the hit attribute.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+*/
+static int max22200_iio_read_hit(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	uint32_t hit_c, hit_t;
+	int ret;
+
+	ret = max22200_get_ch_hit(desc->max22200_desc, channel->address,
```

Comment:
get_ch_hit expect uin8_t* and you are giving uint32_t*. please use the same type of pointer. see everywhere where applicable